File: Examples/test-suite/perl5/director_enum_runme.pl
use strict;
use warnings;
use director_enum;

{
	package MyFoo;
	use base 'director_enum::Foo';
	sub say_hi { my($self, $val) = @_;
		return $val;
	}
}

my $b = director_enum::Foo->new();
my $a = MyFoo->new();

die "RuntimeError" if $a->say_hi($director_enum::hello) !=
	$a->say_hello($director_enum::hi);

File: Examples/test-suite/php/director_enum_runme.php
<?php

require "tests.php";

// No new functions
check::functions(array());
// New classes
check::classes(array('director_enum','Foo','A','B','A2','B2'));
// No new vars
check::globals(array());

class MyFoo extends Foo {
  function say_hi($val) {
    return $val;
  }
}

$b = new Foo();
$a = new MyFoo();

check::equal($a->say_hi(director_enum::hello), $b->say_hello(director_enum::hi), "say failed");

check::done();
